cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -238,17 +241,13 @@ public void start() {
   /**
    * CDAP-6644 for secure impersonation to work,
    * we want other users to be able to write to the "path" directory,
-   * currently only cdap.user has read-write permissions while other users can only read the "/cdap/{path}" dir,
+   * currently only cdap.user has read-write permissions
+   * while other users can only read the "{hdfs.namespace}/{path}" dir,
    * we want to let others to be able to write to "path" directory, till we have a better solution.
    */
   private void createDirectory(String path) {
-    try {
-      String namespacedPath = String.format("/%s/%s", cConf.get(Constants.ROOT_NAMESPACE), path);
-      FileContext fileContext = FileContext.getFileContext(hConf);
-      createDirectory(fileContext, namespacedPath);
-    }  catch (UnsupportedFileSystemException e) {
-      LOG.error("Unsupported FileSystem Exception while trying to create directory", e);
-    }
+    String namespacedPath = String.format("%s/%s", cConf.get(Constants.CFG_HDFS_NAMESPACE), path);
```

Comment:
So this is the actual code fix, right? I think it's better to stick to the old code, that is without exposing the `FileContext` binding directly through the injector. You should still use the `FileContextProvider.get()`, just not through guice.

```java
FileContext fc = new FileContextProvider(cConf, hConf).get();
...
```